modules/map/hooks/useRegisterDevice.ts

```diff
@@ -0,0 +1,33 @@
+import messaging from '@react-native-firebase/messaging'
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { Platform } from 'react-native'
+
+import { clientApi } from '@/modules/backend/client-api'
+import { devicesOptions } from '@/modules/backend/constants/queryOptions'
+import { MobileDevicePlatform } from '@/modules/backend/openapi-generated'
+
+export const useRegisterDevice = (skipTokenRegistration?: boolean) => {
+  const devicesQuery = useQuery(devicesOptions(skipTokenRegistration))
+
+  const registerDeviceMutation = useMutation({
+    mutationFn: async (tokenInner: string) =>
+      // TODO investigate if/why this request gets resent in infinite loop sometimes
+      clientApi.mobileDevicesControllerInsertMobileDevice({
+        token: tokenInner,
+        platform: Platform.OS === 'ios' ? MobileDevicePlatform.Apple : MobileDevicePlatform.Android,
+      }),
+    onSuccess: async () => {
+      await devicesQuery.refetch()
+    },
+  })
+
+  const registerDeviceIfNotExists = async () => {
+    const token = await messaging().getToken()
+
+    if (token && !devicesQuery.data?.devices.some((device) => device.token === token)) {
```

Comment:
if data are undefined the mutation triggers, do we want it to happen? so if the devicesQuery throws an error this mutation can create a loop we try to avoid.

Reply:
In javascript, `undefined` is "falsy" value, so it does not pass the condition and mutation is not triggered. We you this pattern heavilly in Next/Strapi projects.

See here, "yes" is not logged into browser console when `&& undefined` is used in condition:
![image](https://github.com/bratislava/paas-mpa/assets/19418224/ad1944c1-0608-4b65-a97e-49b44f6d1636)

But we could make it more readable and add  `&& devicesQuery.data` if you think so.

Reply:
<img width="284" alt="Screenshot 2024-06-12 at 12 23 47" src="https://github.com/bratislava/paas-mpa/assets/61841194/a8004f7e-5d6c-4462-acbd-8fd74a6e0c01">
then you are doing it wrong :D

Reply:
Fixed ✅